<?xml version="1.0"?>
<launch>

    <node pkg="carrobot" type="carrobot_node.py" name="carrobot">
      <param name="base_lidar_x" value="0.1"/>
      <param name="base_lidar_y" value="0.0"/>
      <param name="base_lidar_z" value="0.05"/>
      <param name="lidar_motor_pwm_duty_cycle" value="0.9"/>
    </node>

    <node pkg="xv_11_laser_driver" type="neato_laser_publisher" name="neato_laser">
        <param name="~port" value="/dev/ttyUSB0"/>
    </node>

    <arg name="dev" default="/dev/ttyS0"/>
    <arg name="baud" default="38400"/>
    <arg name="address" default="128"/>
    <arg name="max_speed" default="2.0"/>
    <arg name="ticks_per_meter" default="4342.2"/>
    <arg name="base_width" default="0.315"/>
    <arg name="run_diag" default="true"/>

    <node if="$(arg run_diag)" pkg="roboclaw_node" type="roboclaw_node.py" name="roboclaw_node">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address" value="$(arg address)"/>
        <param name="~max_speed" value="$(arg max_speed)"/>
        <param name="~ticks_per_meter" value="$(arg ticks_per_meter)"/>
        <param name="~base_width" value="$(arg base_width)"/>
    </node>

    <node pkg="diagnostic_aggregator" type="aggregator_node"
          name="diagnostic_aggregator">
        <rosparam command="load"
                  file="$(find roboclaw_node)/config/roboclaw_diag.yaml"/>
    </node>

</launch>
